src/1password.ts

```diff
@@ -0,0 +1,178 @@
+import * as core from '@actions/core'
+// import * as exec from '@actions/exec'
+import {install} from './install'
+import * as tc from '@actions/tool-cache'
+import {execWithOutput} from './exec'
+import {Item} from './types'
+import * as parsing from './parsing'
+
+const ONE_PASSWORD_VERSION = '1.8.0'
+
+export class OnePassword {
+  onePasswordEnv: {[key: string]: string}
+
+  constructor(deviceId: string) {
+    this.onePasswordEnv = {
+      OP_DEVICE: deviceId
+    }
+
+    if (process.env['XDG_CONFIG_HOME'] === undefined) {
+      // This env var isn't set on GitHub-hosted runners
+      this.onePasswordEnv.XDG_CONFIG_HOME = `${process.env['HOME']}/.config`
+    }
+  }
+
+  async setupAndInstallIfNeeded(): Promise<void> {
+    // Check if op is installed and download if necessary
+    const cachedOpDirectory = tc.find('op', ONE_PASSWORD_VERSION)
+    // This seems like a weird API, why not return undefined?
+    if (cachedOpDirectory !== '') {
+      core.addPath(cachedOpDirectory)
+    } else {
+      await install(ONE_PASSWORD_VERSION)
+    }
+  }
+
+  async signIn(
+    signInAddress: string,
+    emailAddress: string,
+    secretKey: string,
+    masterPassword: string
+  ): Promise<void> {
+    const env = this.onePasswordEnv
+    try {
+      const output = await execWithOutput(
+        'op',
+        [
+          'signin',
+          signInAddress,
+          emailAddress,
+          secretKey,
+          '--raw',
+          '--shorthand',
+          'github_action'
+        ],
+        {
+          env,
+          input: Buffer.alloc(masterPassword.length, masterPassword)
+        }
+      )
+      core.info('Successfully logged into 1Password')
```

Comment:
```suggestion
      core.info('Successfully signed in to 1Password')
```
just for consistency with op's verb

Reply:
👍